page_separator.py

```diff
@@ -1,11 +1,27 @@
-from PyPDF2 import PdfFileWriter, PdfFileReader
 import argparse
+import sys
+from enum import Enum
+
+from PyPDF2 import PdfFileWriter, PdfFileReader
 from PyPDF2.pdf import PageObject
 
-def is_letter(page: PageObject) -> bool:
-    if page.mediaBox.getHeight() == 792:
-        return True
-    return False
+LETTER_RATIO = 11 / 8.5
+LEGAL_RATIO = 14 / 8.5
+MARGIN_OF_ERROR = 0.05 # page ratio should differ by at most this percent
+
+class PageType(Enum):
+    UNKNOWN = 0
+    LETTER = 1
+    LEGAL = 2
+    
+
+def page_type(page: PageObject) -> PageType:
+    ratio = page.mediaBox.getHeight() / page.mediaBox.getWidth()
+    if abs(LETTER_RATIO - ratio) < MARGIN_OF_ERROR:

```

Comment:
If you want it to be percentage, you need to divide by LETTER_RATIO inside the abs.

Reply:
like this?
```suggestion
    if abs((LETTER_RATIO - ratio)/LETTER_RATIO) < MARGIN_OF_ERROR:
```

Reply:
Yes, but for the LEGAL_RATIO as well.